src/daft-core/src/array/ops/cast.rs

```diff
@@ -1597,7 +1597,9 @@ fn cast_sparse_to_dense_for_inner_dtype(
                     continue;
                 }
                 let index_series: Series = non_zero_indices_array.get(i).unwrap();
-                let index_array = index_series.u64().unwrap().as_arrow();
+                let index_array = index_series.downcast::<<$T as DaftDataType>::ArrayType>()
```

Comment:
I believe this is the value type and not the index type. a workaround here to always cast to uint64 for the check?

```
let index_array = index_series.cast(Datatype::UInt64).u64().unwrap().as_arrow();
```